src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -132,9 +134,8 @@ private function preRender(MountedComponent $mounted, array $context = []): PreR
 
             // add attributes
             [$metadata->getAttributesVar() => $mounted->getAttributes()],
-
-            // expose public properties and properties marked with ExposeInTemplate attribute
-            iterator_to_array($this->exposedVariables($component, $metadata->isPublicPropsExposed())),
+            $props,
+            ['props' => $props]
```

Comment:
So `props` is basically an internal key so that we can decide the correct thing in `PropsNode`. Call it `__props` to make it look more internal?